xchange-lgo/src/main/java/org/knowm/xchange/lgo/service/LgoMarketDataServiceRaw.java

```diff
@@ -0,0 +1,38 @@
+package org.knowm.xchange.lgo.service;
+
+import org.knowm.xchange.lgo.LgoErrorAdapter;
+import org.knowm.xchange.lgo.LgoExchange;
+import org.knowm.xchange.lgo.dto.currency.LgoCurrencies;
+import org.knowm.xchange.lgo.dto.product.LgoProducts;
+import si.mazi.rescu.HttpStatusIOException;
+
+public class LgoMarketDataServiceRaw extends LgoBaseService {
+
+  public LgoMarketDataServiceRaw(LgoExchange exchange) {
+    super(exchange);
+  }
+
+  public LgoProducts getProducts() {
+    LgoProducts products = null;
+    try {
+      products =
+          this.proxy.getProducts(
+              exchange.getNonceFactory().createValue(), exchange.getSignatureService());
+    } catch (HttpStatusIOException e) {
```

Comment:
please improve error handling in accordance to: https://github.com/knowm/XChange/wiki/New-Implementation-Best-Practices#step-7

you should introduce and LgoException that implements HttpStatusExceptionSupport

try catch adapt should be done in method that implement generic interfaces, *Raw methods should throw our the LgoException

Reply:
👍